### ISSUE TITLE
[Bug]: Incorrect handling of video if name changes during metadata refresh

### ISSUE THREAD

**Issue by lue30499** [tubearchivist/tubearchivist#424]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read through the [wiki](https://github.com/tubearchivist/tubearchivist/wiki) and the [readme](https://github.com/tubearchivist/tubearchivist#installing-and-updating), particularly the [common errors](https://github.com/tubearchivist/tubearchivist#common-errors) section.

### Operating System

Linux

### Your Bug Report

Hello,

Not sure what causes it. But it seems as though if a video has a name change after the video is downloaded the web ui isn't able to find the video and gets a 404. 

Example with a Mustard video i recently downloaded:

<img width="1680" alt="Screenshot 2023-02-05 at 11 17 13 pm" src="https://user-images.githubusercontent.com/63104612/216818238-d5cbbf76-a98d-4358-a5fd-e23ece9a04b4.png">

And it looks like this in the filesystem (as you can see, video name is different than the .vtt file):

<img width="1076" alt="Screenshot 2023-02-05 at 11 17 44 pm" src="https://user-images.githubusercontent.com/63104612/216818253-a5b4e8b9-2757-4e98-a421-248a6405b5f8.png">

--

Is there any way I can fix this? I do not have a complete list of what videos have had this happen to them, but I have stumbled on a few and it would be quite tedious to manually fix them as my library is large-ish...

Note: System is running on linux, im just viewing the data via a share on my Mac.

Regards,
Lue.

### Relevant log output

```shell
N/A - Logs are very noisy with comments downloading and I didn't see any errors.
```

### Anything else?

N/A

**Comment by PhuriousGeorge** [comment]
Thank you for the bug report, I'm surprised I hadn't noticed this one myself yet.  I just so happened to watch that video on YT instead of TA yesterday :)

**Comment by bbilly1** [comment]
Yeah, I see how this is a problem, thanks for reporting. The Refresh Metadata task picks up the title change, but doesn't apply it to the media file. I remember dealing with that before, but it looks like it got lost while refactoring. The Subtitle will get the new title, because subtitle files will get recreated during the refresh.

For a quick fix you can run the *Rescan Filesystem* task from the settings page: https://github.com/tubearchivist/tubearchivist/wiki/Settings#rescan-filesystem

That will rename the media file correctly and update it in the DB if needed.

But this still needs handling during the refresh task.

**Comment by lue30499** [comment]
Ah yes thanks, a rescan of filesystem has "fixed" it. Glad I don't have to rename manually 😂

**Comment by bbilly1** [comment]
Thanks for reporting, fix is now published in v0.3.4, please update, and rescan your filesystem if you are affected.
